docs/specs/user-export.md

```diff
@@ -0,0 +1,284 @@
+  - `csv`: The output is a CSV file. See https://datatracker.ietf.org/doc/html/rfc4180
+- `csv.fields`: Required when `format` is `csv`. It must be an non-empty array.
+  - `csv.fields.json_pointer`: Required. Select which field in the record to output. It must be a JSON pointer. See https://datatracker.ietf.org/doc/html/rfc6901 and [The record format](#the-record-format)
+  - `csv.fields.field_name`: Optional. If it is not given, then the field name is derived from `json_pointer` by taking the last reference token. For example, if the JSON pointers are `/sub` and `/address/formatted`, then the field names are `sub` and `formatted` respectively. If the field names are not unique, it is an error when the export is initiated. An error is immediately returned in this case, the import job is not created. See [The error response of Create an export](#the-error-response-of-create-an-export)
+
+### The response body of Create an export
+
+See [The response body](#the-response-body).
+
+### The error response of Create an export
+
+> we have a global middleware of handling Admin API authentication,
+> that middleware returns 403 without a JSON body when authentication failed.
+
+|Description|Name|Reason|Info|
+|---|---|---|---|
+|When the rate limit exceeded|`TooManyRequest`|`RateLimited`|{"bucket_name": "UserExport"}|
+|When there is a running export|`TooManyRequest`|`MaximumConcurrentJobLimitExceeded`|-|
+|When the input fails the validation|`Invalid`|`ValidationFailed`|The info should contain the JSON schema validation output|
+|When the field names are not unique|`Invalid`|`UserExportNonUniqueFieldNames`|Output the full list of "field_names". Like `{"field_names": ["sub", "a", "b", "a"]}`|
+
+## Get the status of an export
+
+The endpoint is `GET /_api/admin/users/export/{ID}`
+
+- The endpoint requires Admin API JWT token to access.
+- This endpoint is added to Admin API server.
+- It is not part of the GraphQL API.
+- The result of a completed export lasts for 24h before it expires.
+- When the export is completed, the response body includes a freshly signed URL to the export file. The signed URL is valid for 60s.
+
+### The response body of Get the status of an export
+
+See [The response body](#the-response-body).
+
+### The error response of Get the status of an export
+
+> we have a global middleware of handling Admin API authentication,
+> that middleware returns 403 without a JSON body when authentication failed.
+
+|Description|Name|Reason|Info|
+|---|---|---|---|
+|When the given ID does not refer to an export|`NotFound`|`TaskNotFound`|-|
+
+## The response body
+
+The response body of a just created export looks like
+
+```
+{
+  "id": "some_opaque_string",
+  "status": "pending",
+  "created_at": "2024-01-01T00:00:00.000Z",
+  "request": {
+    "format": "ndjson"
+  }
+}
+```
+
+The response body of completed export looks like
+
+```
+{
+  "id": "some_opaque_string",
+  "status": "completed",
```

Comment:
I think there should be a `failed` status

Reply:
will we want to store the fail reason as well in the response?